integration/nwo/token/fabric/cc/tcc.go

```diff
@@ -17,20 +17,139 @@ import (
 	"strings"
 	"text/template"
 
-	pp2 "github.com/hyperledger-labs/fabric-token-sdk/cmd/tokengen/cobra/pp/cc"
-
 	"github.com/hyperledger-labs/fabric-smart-client/integration/nwo/fabric/packager"
 	"github.com/hyperledger-labs/fabric-smart-client/integration/nwo/fabric/topology"
-	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/hash"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/flogging"
+	pp2 "github.com/hyperledger-labs/fabric-token-sdk/cmd/tokengen/cobra/pp/cc"
+	common2 "github.com/hyperledger-labs/fabric-token-sdk/integration/nwo/token/common"
+	"github.com/hyperledger-labs/fabric-token-sdk/integration/nwo/token/fabric"
 	topology3 "github.com/hyperledger-labs/fabric-token-sdk/integration/nwo/token/topology"
 	. "github.com/onsi/gomega"
 )
 
+const (
+	DefaultTokenChaincode                    = "github.com/hyperledger-labs/fabric-token-sdk/token/services/network/fabric/tcc/main"
+	DefaultTokenChaincodeParamsReplaceSuffix = "/token/services/network/fabric/tcc/params.go"
+)
+
+var logger = flogging.MustGetLogger("token-sdk.integration.token.fabric.cc")
+
+type fabricPlatform interface {
+	Topology() *topology.Topology
+	UpdateChaincode(name string, version string, path string, file string)
+}
+
 type TCC struct {
 	Chaincode *topology.ChannelChaincode
 }
 
-func (p *NetworkHandler) tccSetup(tms *topology3.TMS, cc *topology.ChannelChaincode) (*topology.ChannelChaincode, uint16) {
+type GenericBackend struct {
+	TokenChaincodePath                string
+	TokenChaincodeParamsReplaceSuffix string
+	TokenPlatform                     common2.TokenPlatform
+}
+
+func NewDefaultGenericBackend(tokenPlatform common2.TokenPlatform) *GenericBackend {
+	return NewGenericBackend(
+		DefaultTokenChaincode,
+		DefaultTokenChaincodeParamsReplaceSuffix,
+		tokenPlatform,
+	)
+}
+
+func NewGenericBackend(tokenChaincodePath string, tokenChaincodeParamsReplaceSuffix string, tokenPlatform common2.TokenPlatform) *GenericBackend {
+	return &GenericBackend{TokenChaincodePath: tokenChaincodePath, TokenChaincodeParamsReplaceSuffix: tokenChaincodeParamsReplaceSuffix, TokenPlatform: tokenPlatform}
+}
+
+func (p *GenericBackend) PrepareNamespace(entry *fabric.Entry, tms *topology3.TMS) {
```

Comment:
Do we need the entry as a parameter? If another backend comes that needs the entry as input, we can add it later to the interface.

Reply:
okay